src/main/java/org/mitre/synthea/export/FhirDstu2.java

```diff
@@ -1362,6 +1371,72 @@ private static Entry imagingStudy(Entry personEntry, Bundle bundle, Entry encoun
     return newEntry(bundle, imagingStudyResource);
   }
 
+  /**
+   * Map the HealthRecord.Device into a FHIR Device and add it to the Bundle.
+   *
+   * @param personEntry    The Person entry.
+   * @param bundle         Bundle to add to.
+   * @param device         The device to add.
+   * @return The added Entry.
+   */
+  private static Entry device(Entry personEntry, Bundle bundle,
+      HealthRecord.Device device) {
+    Device deviceResource = new Device();
+
+    deviceResource.setUdi(device.udi);
+    deviceResource.setStatus(DeviceStatusEnum.AVAILABLE);
+    if (device.manufacturer != null) {
+      deviceResource.setManufacturer(device.manufacturer);
+    }
+    if (device.model != null) {
+      deviceResource.setModel(device.model);
+    }
+    deviceResource.setManufactureDate((DateTimeDt)convertFhirDateTime(device.manufactureTime, true));
+    deviceResource.setExpiry((DateTimeDt)convertFhirDateTime(device.expirationTime, true));
+    deviceResource.setLotNumber(device.lotNumber);
+    deviceResource.setType(mapCodeToCodeableConcept(device.codes.get(0), SNOMED_URI));
+    deviceResource.setPatient(new ResourceReferenceDt(personEntry.getFullUrl()));
+
+    return newEntry(bundle, deviceResource);
+  }
+  
+  /**
+   * Map the JsonObject for a Supply into a FHIR SupplyDelivery and add it to the Bundle.
+   *
+   * @param personEntry    The Person entry.
+   * @param bundle         Bundle to add to.
+   * @param supply         The supplied object to add.
+   * @param encounter      The encounter during which the supplies were delivered
+   * @return The added Entry.
+   */
+  private static Entry supplyDelivery(Entry personEntry, Bundle bundle,
+      HealthRecord.Supply supply, Encounter encounter) {
+   
+    SupplyDelivery supplyResource = new SupplyDelivery();
+    supplyResource.setStatus(SupplyDeliveryStatusEnum.DELIVERED);
+    supplyResource.setPatient(new ResourceReferenceDt(personEntry.getFullUrl()));
+
+    CodeableConceptDt type = new CodeableConceptDt();
+    type.addCoding()
+      .setCode("device")
+      .setDisplay("Device")
+      .setSystem(SNOMED_URI);
```

Comment:
I think it is an error to use `SNOMED_URI` for `SupplyDelivery.type` -- you should use `http://terminology.hl7.org/CodeSystem/supply-item-type` in FHIR R4 (see http://hl7.org/fhir/valueset-supplydelivery-type.html#expansion) and `http://hl7.org/fhir/supply-item-type` in FHIR STU3 (see http://hl7.org/fhir/STU3/valueset-supplydelivery-type.html#expansion) and `http://hl7.org/fhir/supply-item-type` in FHIR DSTU2 (see http://hl7.org/fhir/DSTU2/valueset-supplydelivery-type.html#definition)